hypothesis-python/tests/conjecture/test_alt_backend.py

```diff
@@ -718,3 +719,45 @@ def test_replay_choices():
     # trivial covering test
     provider = TrivialProvider(None)
     provider.replay_choices([1])
+
+
+class ObservationProvider(TrivialProvider):
+    add_observability_callback = True
+
+    def __init__(self, conjecturedata: "ConjectureData", /) -> None:
+        super().__init__(conjecturedata)
+        # calls to per_test_case_context_manager and on_observation alternate,
+        # starting with per_test_case_context_manager
+        self.expected = "per_test_case_context_manager"
+
+    @contextmanager
+    def per_test_case_context_manager(self):
+        assert self.expected == "per_test_case_context_manager"
+        self.expected = "on_observation"
+        yield
+
+    def on_observation(self, observation: Observation) -> None:
+        assert self.expected == "on_observation"
+        self.expected = "per_test_case_context_manager"
+
+
+def test_on_observation_alternates():
+    with temp_register_backend("observation", ObservationProvider):
```

Comment:
Neat trick: functions decorated with `@contextmanager` can also be used as decorators, e.g.
```suggestion
@temp_register_backend("observation", ObservationProvider)
def test_on_observation_alternates():
```